actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -143,24 +144,25 @@ def escape_once(html)
         def content_tag_string(name, content, options, escape = true)
           tag_options = tag_options(options, escape) if options
           content     = ERB::Util.unwrapped_html_escape(content) if escape
-          "<#{name}#{tag_options}>#{PRE_CONTENT_STRINGS[name.to_sym]}#{content}</#{name}>".html_safe
+          "<#{name}#{tag_options}>#{PRE_CONTENT_STRINGS[name]}#{content}</#{name}>".html_safe
         end
 
         def tag_options(options, escape = true)
           return if options.blank?
-          attrs = []
+          output = ""
+          sep    = " ".freeze
           options.each_pair do |key, value|
             if TAG_PREFIXES.include?(key) && value.is_a?(Hash)
               value.each_pair do |k, v|
-                attrs << prefix_tag_option(key, k, v, escape)
+                output << sep + prefix_tag_option(key, k, v, escape)
```

Comment:
I was of the understanding that `sep + prefix_tag_option(key, k, v, escape)` would allocate a new string object. This could be avoided by shoveling twice instead of once:
```rb
output << sep
output << prefix_tag_option(key, k, v, escape)
```